src/main/java/slimeknights/tconstruct/library/modifiers/modules/technical/MaxArmorLevelModule.java

```diff
@@ -0,0 +1,102 @@
+package slimeknights.tconstruct.library.modifiers.modules.technical;
+
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.tags.TagKey;
+import net.minecraft.world.entity.EquipmentSlot;
+import net.minecraft.world.entity.EquipmentSlot.Type;
+import net.minecraft.world.item.Item;
+import slimeknights.mantle.data.loadable.Loadables;
+import slimeknights.mantle.data.loadable.primitive.BooleanLoadable;
+import slimeknights.mantle.data.loadable.record.RecordLoadable;
+import slimeknights.tconstruct.library.modifiers.ModifierEntry;
+import slimeknights.tconstruct.library.modifiers.ModifierHooks;
+import slimeknights.tconstruct.library.modifiers.data.ModifierMaxLevel;
+import slimeknights.tconstruct.library.modifiers.hook.armor.EquipmentChangeModifierHook;
+import slimeknights.tconstruct.library.modifiers.modules.ModifierModule;
+import slimeknights.tconstruct.library.module.HookProvider;
+import slimeknights.tconstruct.library.module.ModuleHook;
+import slimeknights.tconstruct.library.tools.capability.TinkerDataCapability;
+import slimeknights.tconstruct.library.tools.capability.TinkerDataCapability.TinkerDataKey;
+import slimeknights.tconstruct.library.tools.context.EquipmentChangeContext;
+import slimeknights.tconstruct.library.tools.nbt.IToolStackView;
+import slimeknights.tconstruct.library.utils.IdParser;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+/**
+ * Module for keeping track of the max level of a modifier across all pieces of equipment.
+ * @see TinkerDataKey
+ * @see slimeknights.tconstruct.library.modifiers.modules.behavior.ShowOffhandModule
+ */
+public record MaxArmorLevelModule(TinkerDataCapability.ComputableDataKey<ModifierMaxLevel> key, boolean allowBroken, @Nullable TagKey<Item> heldTag) implements HookProvider, EquipmentChangeModifierHook, ModifierModule {
+  private static final List<ModuleHook<?>> DEFAULT_HOOKS = HookProvider.<MaxArmorLevelModule>defaultHooks(ModifierHooks.EQUIPMENT_CHANGE);
+  public static final RecordLoadable<MaxArmorLevelModule> LOADER = RecordLoadable.create(
+    new IdParser<>(ResourceLocation::new, "Stat").flatXmap(location -> TinkerDataCapability.ComputableDataKey.of(location, ModifierMaxLevel::new), TinkerDataKey::getId).requiredField("key", MaxArmorLevelModule::key),
```

Comment:
Why is "Stat" capitalized? All the other fields are lowercase.

That said, given this key is created specifically for this module, it won't be possible to use it in code so there is currently no point in adding this module. I'd recommend making this just implement `HookProvider` instead of `ModifierModule` for now and removing its loader. We can always add it back later with a `ModifierMaxLevel` registry if needed.